core/src/main/scala/org/apache/spark/sql/SnappySession.scala

```diff
@@ -1949,7 +1949,7 @@ object SnappySession extends Logging {
         cachedDF = evaluation._1
         queryHints = evaluation._2
       } else {
-        cachedDF.clearCachedShuffleDeps(session.sparkContext)
+        cachedDF.waitForLastShuffleCleanup
```

Comment:
convention is that for actions the empty parens should be provided (and skipped for getters)